internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,174 @@
+	}
+}
+
+func (d *daemonset) SetTopologyDaemonsetAsDesired(ctx context.Context, nfdInstance *nfdv1.NodeFeatureDiscovery, topologyDS *appsv1.DaemonSet) error {
+	topologyDS.ObjectMeta.Labels = map[string]string{"app": "nfd"}
+
+	podLabels := map[string]string{"app": "nfd-topology-updater"}
+	topologyDS.Spec = appsv1.DaemonSetSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: podLabels,
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: podLabels,
+			},
+			Spec: corev1.PodSpec{
+				ServiceAccountName: "nfd-topology-updater",
+				DNSPolicy:          corev1.DNSClusterFirstWithHostNet,
+				Containers: []corev1.Container{
+					{
+						Name:            "nfd-topology-updater",
+						Image:           nfdInstance.Spec.Operand.ImagePath(),
+						ImagePullPolicy: getImagePullPolicy(nfdInstance),
+						Command: []string{
+							"nfd-topology-updater",
+						},
+						Args:            getArgs(nfdInstance),
+						Env:             getEnvs(),
+						SecurityContext: getSecurityContext(),
+						VolumeMounts:    getVolumeMounts(),
+					},
+				},
+				Volumes: getVolumes(),
+			},
+		},
+	}
+	return controllerutil.SetControllerReference(nfdInstance, topologyDS, d.scheme)
+}
+
+func getImagePullPolicy(nfdInstance *nfdv1.NodeFeatureDiscovery) corev1.PullPolicy {
+	if nfdInstance.Spec.Operand.ImagePullPolicy != "" {
+		return corev1.PullPolicy(nfdInstance.Spec.Operand.ImagePullPolicy)
+	}
+	return corev1.PullAlways
+}
+
+func getArgs(nfdInstance *nfdv1.NodeFeatureDiscovery) []string {
+	return []string{
+		"-kubelet-config-uri=file:/host-var/lib/kubelet/config.yaml",
+		"-podresources-socket=/host-var/lib/kubelet/pod-resources/kubelet.sock",
+		"-sleep-interval=3s",
+	}
+}
+
+func getEnvs() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name: "NODE_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "spec.nodeName",
+				},
+			},
+		},
+	}
+}
+
+func getSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		RunAsUser: ptr.To[int64](0),
+		SELinuxOptions: &corev1.SELinuxOptions{
+			Type: "container_runtime_t",
+		},
+		ReadOnlyRootFilesystem: ptr.To(true),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{"ALL"},
+		},
+		AllowPrivilegeEscalation: ptr.To(true),
+	}
+}
+
+func getVolumeMounts() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			Name:      "kubelet-podresources-conf",
+			MountPath: "/host-var/lib/kubelet/config.yaml",
+			ReadOnly:  true,
+		},
+		{
+			Name:      "kubelet-podresources-sock",
+			MountPath: "/host-var/lib/kubelet/pod-resources/kubelet.sock",
+		},
+		{
+			Name:      "host-sys",
+			MountPath: "/host-sys",
+		},
+	}
+}
+
+func getVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		{
+			Name: "kubelet-podresources-conf",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/kubernetes/kubelet.conf",
```

Comment:
IF we want to use the config file (instead of configz which is the default), I think we should mount `/var/lib/kubelet/config.yaml`. Isn't `/etc/kubernetes/kubelet.conf` a kubeconfig file (for kubelet) which we are not interested in(?)

Reply:
currently we will go with the bare minimum, if the configz defined in the main as default, we won't provide  the option to reconfigure. This can all be done at the later point, when  a need arises